### ISSUE TITLE
trailing white space in "-----BEGIN CERTIFICATE----- " causes readPemObject returning null

### ISSUE THREAD

**Issue by WalkerWalker** [bcgit/bc-java#1533]
if the certificate starts with "-----BEGIN CERTIFICATE----- ", or in general any pem with the begining line including a trailing white space,  the readPemObject() function would return null and this causes the readObject in PEMParser to also return null.

It is because this line of code https://github.com/bcgit/bc-java/blob/1cae5431b0f315e1c7dc40392bcd60cf785831b4/core/src/main/java/org/bouncycastle/util/io/pem/PemReader.java#L46

I wonder if this is design choice or not?   There are some old discussions about whether trailing white space(s) are allowed in x509certificate https://stackoverflow.com/questions/26496155/are-spaces-alowed-in-x509-certificates but it seems to me that the answers are not clear.

**Comment by dghgit** [comment]
I'll admit it is a bit of a weird one, our parser was originally based on RFC 1421, the grammar at the end of it gives:

```
   <preeb> ::= "-----BEGIN PRIVACY-ENHANCED MESSAGE-----" CRLF
   <posteb> ::= "-----END PRIVACY-ENHANCED MESSAGE-----" CRLF / <preeb>
```
So at least as far as the original PEM definition goes, trailing white space would be an error. RFC 7468 is more recent and appears somewhat more forgiving though - the grammar in section 3 doesn't require CRLF and the standard does specifically talk about ignore whitespace. I think we can accommodate a line.trim() prior to doing the checks. I'll put it on the list.

**Comment by dghgit** [comment]
This is now done in 178b02 - see https://www.bouncycastle.org/betas for details.

**Comment by WalkerWalker** [comment]
Thanks for the support.  I wonder if 178b02 is included in 1.78 and/or 1.78.1 published in April, or not? 

> This is now done in 178b02 - see https://www.bouncycastle.org/betas for details.
